src/client/entity-editor/validators/base.js

```diff
@@ -96,41 +94,47 @@ export function validateBoolean(
 	return _.isBoolean(value);
 }
 
-export function validateDate(
-	value: mixed, required: boolean = false
-): boolean {
-	if (absentAndRequired(value, required)) {
-		return false;
-	}
-
-	if (!nilOrString(value)) {
-		return false;
-	}
+export function validateDate(value: mixed) {
```

Comment:
Here is a way to get the validateDate and isNullValue  to accept all kinds of values, including nulls and empty objects:

```
export function validateDate(value: mixed) {
	const year = _.get(value, 'year');
	const month = _.get(value, 'month');
	const day = _.get(value, 'day');
	const {isValid, errorMessage} = dateValidator(day, month, year);
	return {errorMessage, isValid};
}

export function isNullValue(date) {
	return !_.get(date, 'day') && !_.get(date, 'month') && !_.get(date, 'year');
}
```